internal/logger/logger_test.go

```diff
@@ -0,0 +1,283 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"bytes"
+	"log/slog"
+	"strings"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/config"
+	. "github.com/jacobsa/ogletest"
+)
+
+const (
+	textTraceString   = "severity=TRACE msg=\"TestLogs: www.traceExample.com\""
+	textDebugString   = "severity=DEBUG msg=\"TestLogs: www.debugExample.com\""
+	textInfoString    = "severity=INFO msg=\"TestLogs: www.infoExample.com\""
+	textWarningString = "severity=WARNING msg=\"TestLogs: www.warningExample.com\""
+	textErrorString   = "severity=ERROR msg=\"TestLogs: www.errorExample.com\""
+	jsonTraceString   = "\"severity\":\"TRACE\",\"msg\":\"TestLogs: www.traceExample.com\"}"
+	jsonDebugString   = "\"severity\":\"DEBUG\",\"msg\":\"TestLogs: www.debugExample.com\"}"
+	jsonInfoString    = "\"severity\":\"INFO\",\"msg\":\"TestLogs: www.infoExample.com\"}"
+	jsonWarningString = "\"severity\":\"WARNING\",\"msg\":\"TestLogs: www.warningExample.com\"}"
+	jsonErrorString   = "\"severity\":\"ERROR\",\"msg\":\"TestLogs: www.errorExample.com\"}"
+)
+
+func TestLogger(t *testing.T) { RunTests(t) }
+
+type LoggerTest struct {
+}
+
+func init() { RegisterTestSuite(&LoggerTest{}) }
+
+// //////////////////////////////////////////////////////////////////////
+// Boilerplate
+// //////////////////////////////////////////////////////////////////////
+func testLogger(level config.LogSeverity, functions []func()) []string {
+	// create a logger that writes to buffer at configured level.
+	var buf bytes.Buffer
+	var programLevel = new(slog.LevelVar)
+	logger := slog.New(
+		defaultLoggerFactory.createJsonOrTextHandler(&buf, programLevel, "TestLogs: "),
+	)
+	setLoggingLevel(level, programLevel)
+
+	// make the created logger default.
+	defaultLogger = logger
+
+	var output []string
+	// run the functions provided.
+	for _, f := range functions {
+		f()
+		output = append(output, buf.String())
+		buf.Reset()
+	}
+	return output
+}
+
+func getTestFunctions() []func() {
+	return []func(){
+		func() {
+			Tracef("www.traceExample.com")
+		},
+		func() {
+			Debugf("www.debugExample.com")
+		},
+		func() {
+			Infof("www.infoExample.com")
+		},
+		func() {
+			Warnf("www.warningExample.com")
+		},
+		func() {
+			Errorf("www.errorExample.com")
+		},
+	}
+}
+
+func validateOutput(expected []string, output []string) {
+	for i := range output {
+		if expected[i] == "" {
+			AssertEq(expected[i], output[i])
+		} else {
+			AssertTrue(strings.Contains(output[i], expected[i]))
+		}
+	}
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Tests
+// //////////////////////////////////////////////////////////////////////
+func (t *LoggerTest) TestTextFormatLogs_LogLevelOFF() {
```

Comment:
For all the tests, please separate `arrange`, `act` and `assert` block by inserting a line.